packages/sql/src/Client.ts

```diff
@@ -0,0 +1,99 @@
+import { DEFAULT_SERVER, PUBLIC_API_KEY } from './constants';
+import { CopyFromManager } from './CopyFromManager';
+import { CopyToManager } from './CopyToManager';
+import DDL, { ColumConfig, CreateConfig, DropOptions } from './DDL';
+import { QueryManager } from './QueryManager';
+
+export class SQL {
+  private _copyToManager: CopyToManager;
+  private _queryManager: QueryManager;
+  private _copyFromManager: CopyFromManager;
+  private _publicQueryManager: QueryManager;
+  private _publicRole?: string;
+
+  constructor(username: string, apiKey: string, server: string = DEFAULT_SERVER) {
```

Comment:
should we use Credentials here too? Having an object could be useful to add more (optional) properties in the future.

I'm currently using (at maps-api) `constructor(username: string, apiKey: string, serverURL?: string, client?: string)` 

A client set to  'toolkit-maps' / 'toolkit-sql' could be useful to get some metrics

Reply:
I think we should use something else, probably something that the auth package can provide, but not necessarily as a real dependency.

An object with the shape of those arguments would work.

I think we should tackle this after we're done, and solve the issue of updating the API key by adding a setter _for now_

Reply:
yeah, for sure, let's move on. I just wanted to point out the need of thinking more about this common pieces / repetitions